pages/api/auth/[...nextauth].js

```diff
@@ -0,0 +1,35 @@
+import NextAuth from "next-auth";
+import GithubProvider from "next-auth/providers/github";
+import CredentialsProvider from "next-auth/providers/credentials";
+
+export const authOptions = {
+  providers: [
+    process.env.VERCEL_ENV === "preview"
+      ? CredentialsProvider({
+          name: "Credentials",
+          credentials: {
+            username: {
+              label: "Username",
+              type: "text",
+              placeholder: "newfishes",
+            },
+            password: { label: "Password", type: "password" },
+          },
+          async authorize() {
+            return {
+              id: 1,
+              name: "New Fishes",
+              email: "new@fishes.com",
+              image:
+                "https://avatars.githubusercontent.com/u/41701911?s=200&v=4",
+            };
+          },
+        })
+      : GithubProvider({
+          clientId: process.env.GITHUB_ID,
+          clientSecret: process.env.GITHUB_SECRET,
+        }),
+  ],
```

Comment:
please make the github signin in the preview mode working on vercel. You can use the "branch" link to your application then you don't need to update the callback link with every commit. If you cant find it let me know.